bundles/org.openhab.binding.gpio/src/main/java/org/openhab/binding/gpio/handler/GPIODigitalInputHandler.java

```diff
@@ -0,0 +1,189 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.gpio.handler;
+
+import static org.openhab.binding.gpio.GPIOBindingConstants.*;
+
+import java.math.BigDecimal;
+import java.util.Date;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import eu.xeli.jpigpio.Alert;
+import eu.xeli.jpigpio.GPIO;
+import eu.xeli.jpigpio.JPigpio;
+import eu.xeli.jpigpio.PigpioException;
+
+/**
+ * Thing Handler for digital GPIO unputs.
+ *
+ * @author Nils Bauer - Initial contribution
+ */
+public class GPIODigitalInputHandler extends BaseThingHandler {
+
+    /** The logger. */
+    private Logger logger = LoggerFactory.getLogger(GPIODigitalInputHandler.class);
+
+    /** The JPigpio instance. */
+    private JPigpio jPigpio;
+
+    /** The gpio pin handled by this handler. */
+    private GPIO gpio;
+
+    /** The last change. */
+    private Date lastChanged;
+
+    /**
+     * Instantiates a new GPIO digital input handler.
+     *
+     * @param thing the thing
+     */
+    public GPIODigitalInputHandler(Thing thing) {
+        super(thing);
+    }
+
+    /*
+     * (non-Javadoc)
+     *
+     * @see
+     * org.eclipse.smarthome.core.thing.binding.ThingHandler#handleCommand(org.eclipse.smarthome.core.thing.ChannelUID,
+     * org.eclipse.smarthome.core.types.Command)
+     */
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        try {
+            Channel channel = getThing().getChannel(THING_TYPE_DIGITAL_INPUT_CHANNEL);
+            if (channel == null) {
+                logger.error("Cannot find Channel: " + THING_TYPE_DIGITAL_INPUT_CHANNEL);
```

Comment:
Please don't log to error. Within openHAB error is used for reporting system wide failure reporting. Use warn or debug depending how important the message is to see in case it's a problem.